file/src/main/java/org/apache/calcite/adapter/file/CsvEnumerator.java

```diff
@@ -130,30 +129,108 @@ public static RelDataType deduceRowType(JavaTypeFactory typeFactory,
       }
       for (String string : strings) {
         final String name;
-        final CsvFieldType fieldType;
+        final RelDataType fieldType;
         final int colon = string.indexOf(':');
         if (colon >= 0) {
           name = string.substring(0, colon);
           String typeString = string.substring(colon + 1);
-          fieldType = CsvFieldType.of(typeString);
-          if (fieldType == null) {
-            System.out.println("WARNING: Found unknown type: "
-                + typeString + " in file: " + source.path()
-                + " for column: " + name
-                + ". Will assume the type of column is string");
+          Matcher decimalMatcher = DECIMAL_TYPE_PATTERN.matcher(typeString);
+          if (decimalMatcher.matches()) {
+            int precision = Integer.parseInt(decimalMatcher.group(1));
+            int scale = Integer.parseInt(decimalMatcher.group(2));
+            fieldType = typeFactory.createTypeWithNullability(typeFactory
+                .createSqlType(SqlTypeName.DECIMAL,
+                    precision, scale), true);
+          } else {
+            switch (typeString) {
+            case "string":
+              fieldType = typeFactory.createTypeWithNullability(typeFactory
+                      .createSqlType(typeFactory.createJavaType(String.class).getSqlTypeName()),
+                  true);
+              break;
+            case "boolean":
+              fieldType = typeFactory.createTypeWithNullability(
```

Comment:
This pattern seems very redundant. It seems like 99% of the time, the pattern is string => sqlTypeName (everything except decimal). Can your refactor to avoid all the repetition?

Reply:
Refactored to a helper method now.